haystack/nodes/preprocessor/splitter.py

```diff
@@ -0,0 +1,755 @@
+        self.progress_bar = progress_bar
+        self.add_page_number = add_page_number
+
+        self.merger = DocumentMerger(
+            separator="",
+            window_size=split_length,
+            window_overlap=split_overlap,
+            max_tokens=max_tokens,
+            retain_page_number=True,
+            realign_headlines=True,
+        )
+
+        self._nltk_language = nltk_language
+        self._nltk_folder = Path(nltk_folder) if nltk_folder else None
+        self.sentence_tokenizer = load_sentence_tokenizer(
+            language=self._nltk_language, tokenizer_model_folder=self._nltk_folder
+        )
+
+        self._tokenizer = None
+        if tokenizer_model or max_tokens:
+            self.tokenizer = tokenizer_model
+
+        if split_by == "token" and not self.tokenizer:
+            raise ValueError(
+                "If you set split_by='token', you must give a value to 'tokenizer_model'. "
+                "Use the same model you're using for your Reader."
+            )
+
+    def _validate_split_params(
+        self,
+        split_by: SplitBy,
+        split_regex: Optional[str],
+        split_length: int,
+        split_overlap: int,
+        max_chars: int,
+        max_tokens: Optional[int],
+    ):
+        """
+        Performs some basic validation on the parameters of the splitter.
+        """
+        if split_by not in get_args(SplitBy):
+            raise ValueError(f"split_by must be one of: {', '.join(get_args(SplitBy))}")
+
+        if not isinstance(split_length, int) or split_length < 0:
+            raise ValueError("split_length must be an integer >= 0")
+
+        if split_length == 0 and not max_tokens:
+            logger.warning(
+                "split_length is set to 0 and max_tokens is not set. "
+                "This means that the documents will be split in chunks of %s chars. "
+                "Is this the behavior you expect? If so, set `split_by='character'` "
+                "and `split_length=%s` to remove this warning.",
+                max_chars,
+                max_chars,
+            )
+
+        if split_length:
+            if not isinstance(split_overlap, int) or split_overlap < 0:
+                raise ValueError("split_overlap must be an integer >= 0")
+
+            if split_overlap >= split_length:
+                raise ValueError("split_length must be higher than split_overlap")
+
+        if split_regex and not split_by == "regex":
+            logger.warning(
+                "You provided a value to 'split_regex', but 'split_by=\"%s\"'. "
+                "The document will be split by %s and the regex pattern will be ignored.",
+                split_by,
+                split_by,
+            )
+
+    @property
+    def tokenizer(self):
+        return self._tokenizer
+
+    @tokenizer.setter
+    def tokenizer(self, tokenizer_model=Union[str, Path, PreTrainedTokenizer, TokenizerI]):
+        if not tokenizer_model:
```

Comment:
We mention in the error message that the tokenizer can't be set to `None`, but this check will also trigger if the user were to pass an empty string or an empty list as the tokenizer_model as well (e.g. `tokenizer_model = ""`. I think we should either rewrite it as `if tokenizer_model is None` or update the `ValueError` message to return the value provided by the user instead of assuming it was `None`.